### PR TITLE
Add module for monitoring file/folder size like Linux 'du' command

### PR DESCRIPTION
**Pull request by ski2per**
Signed-off-by: Ted <ski2per@gmail.com>

Hi~ We are using Netdata as our production monitoring system. It's really an awesome tool. 
In our production system, we have  requirement to monitor the size for some folders.
I have read Netdata docs and googled for some while, find no solutions.
So I try to write a go plugin for it, wish it would help :)

### CODE REVIEW

File: modules/du/du.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		},
+		Create: func() module.Module { return New() },
+	})
+}
+
+func fileSize(path string) (int64, error) {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
```
Comment (thiagoftsm) [inline modules/du/du.go:40]: Hello @ski2per ,

This function returns two values according its declaration, but here you are returning only one value, please, fix this.

File: modules/du/du.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		}
+		if !info.IsDir() {
+			size += info.Size()
+		}
+		return err
```
Comment (thiagoftsm) [inline modules/du/du.go:45]: The same think is happening here.

File: modules/du/du.go (hunk 1 above)
Comment (thiagoftsm) [reply modules/du/du.go:40]: sorry , I missed the fact this is inside another function.

Comment (ski2per) [reply modules/du/du.go:40]: Wish it would work, i just jump from python to go 🤣

**Comment by ilyam8** [comment]
Hi @ski2per 

`filecheck` collector monitors files size

> https://github.com/netdata/go.d.plugin/tree/master/modules/filecheck#filecheck

As for directories it monitors:
 - existence
 - time since the last modification
 - number of files

I guess we could extend it to monitor summarize disk usage. What do you think?

**Comment by ski2per** [comment]
> Hi @ski2per
> 
> `filecheck` collector monitors files size
> 
> > https://github.com/netdata/go.d.plugin/tree/master/modules/filecheck#filecheck
> 
> As for directories it monitors:
> 
> * existence
> * time since the last modification
> * number of files
> 
> I guess we could extend it to monitor summarize disk usage. What do you think?



> Hi @ski2per
> 
> `filecheck` collector monitors files size
> 
> > https://github.com/netdata/go.d.plugin/tree/master/modules/filecheck#filecheck
> 
> As for directories it monitors:
> 
> * existence
> * time since the last modification
> * number of files
> 
> I guess we could extend it to monitor summarize disk usage. What do you think?

I think I might

> Hi @ski2per
> 
> `filecheck` collector monitors files size
> 
> > https://github.com/netdata/go.d.plugin/tree/master/modules/filecheck#filecheck
> 
> As for directories it monitors:
> 
> * existence
> * time since the last modification
> * number of files
> 
> I guess we could extend it to monitor summarize disk usage. What do you think?

I think I might overlook filecheck module. Extending it should be a good idea. Let me have a try.

I will close this PR

**Comment by ski2per** [comment]
> Hi @ski2per
> 
> `filecheck` collector monitors files size
> 
> > https://github.com/netdata/go.d.plugin/tree/master/modules/filecheck#filecheck
> 
> As for directories it monitors:
> 
> * existence
> * time since the last modification
> * number of files
> 
> I guess we could extend it to monitor summarize disk usage. What do you think?

I think I might overlook the filecheck module. Extending it should be a good idea. Let me have a try.

I will close this PR now.

**Pull request closed without merging**